web/modules/custom/dagplejelager_actions/src/Controller/DagplejelagerActionsController.php

```diff
@@ -0,0 +1,129 @@
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provide actions for Dagplejelager.
+ */
+class DagplejelagerActionsController extends ControllerBase {
+
+  /**
+   * The cart manager.
+   *
+   * @var \Drupal\commerce_cart\CartManagerInterface
+   */
+  protected $cartManager;
+
+  /**
+   * The cart provider.
+   *
+   * @var \Drupal\commerce_cart\CartProviderInterface
+   */
+  protected $cartProvider;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new CartController object.
+   *
+   * @param \Drupal\commerce_cart\CartManagerInterface $cart_manager
+   *   The cart manager.
+   * @param \Drupal\commerce_cart\CartProviderInterface $cart_provider
+   *   The cart provider.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(CartManagerInterface $cart_manager, CartProviderInterface $cart_provider, EntityTypeManagerInterface $entityTypeManager) {
+    $this->cartManager = $cart_manager;
+    $this->cartProvider = $cart_provider;
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('commerce_cart.cart_manager'),
+      $container->get('commerce_cart.cart_provider'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Add a bundle of products to the cart.
+   *
+   * @param \Drupal\commerce_product\Entity\ProductInterface $commerce_product
+   *   The product bundle.
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   The redirect.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function actionAddBundle(ProductInterface $commerce_product, Request $request) {
+    $storeId = $commerce_product->get('stores')->getValue()[0]['target_id'];
+    $store = $this->entityTypeManager()
+      ->getStorage('commerce_store')
+      ->load($storeId);
+
+    /** @var \Drupal\commerce_store\Entity\StoreInterface|void $store */
+    $cart = $this->cartProvider->getCart('default', $store);
+    if (!$cart) {
+      $cart = $this->cartProvider->createCart('default', $store);
+    }
+    if ($commerce_product->hasField('field_product_reference') && !empty($commerce_product->field_product_reference->getValue())) {
+      foreach ($commerce_product->field_product_reference->getValue() as $product_id) {
```

Comment:
By using `referencedEntities` we should be able to get the actual products (and not just the id) and pass these on to `addProductVariationToCart`.